DOM+js+html/ui/script.js

```diff
@@ -0,0 +1,579 @@
+        let post = document.getElementById(id);
+        if(!container.removeChild(post)){
+            return -1;
+        }
+        for(let i = 0; i < photoPosts.length; i++){
+            if(photoPosts[i].id == id){
+                photoPosts[i].isDeleted = true;
+                authorsSelect();
+                hashtagsSelect();
+                return 0;
+            }
+        }
+        // if (!js.removePhotoPost(id)) {
+        //     return -1;
+        // }
+        // showPostsToHtml();
+    }
+
+    function editPostFromHtml(id, change) {
+        if (!js.editPhotoPost(id, change)) {
+            return -1;
+        }
+        hashtagsSelect();
+        showPostsToHtml();
+    }
+
+
+    function authorsSelect() {
+        let set = new Set();
+        for (let i = 0; i < photoPosts.length; i++) {
+            if(photoPosts[i].isDeleted === false) {
+                set.add(photoPosts[i].author);
+            }
+        }
+        let an = document.getElementById("authornames");
+        while (an.firstChild) {
+            an.removeChild(an.firstChild);
+        }
+        set.forEach(function (value) {
+            an.innerHTML += "<option>" + value + "</option>";
+        });
+    }
+
+    function hashtagsSelect() {
+        let set = new Set();
+        for (let i = 0; i < photoPosts.length; i++) {
+            if(photoPosts[i].isDeleted === false) {
+                photoPosts[i].hashtags.forEach(function (value) {
+                    set.add(value);
+                });
+            }
+        }
+        let an = document.getElementById("hashtags");
+        while (an.firstChild) {
+            an.removeChild(an.firstChild);
+        }
+        set.forEach(function (value) {
+            an.innerHTML += "<option>" + value + "</option>";
+        });
+    }
+
+    function addLike(id, author) {
+        if (!js.editPhotoPost(id, {likes: new Set(author)})) {
+            return -1;
+        }
+        let post = document.getElementById(id);
+        if(post === null){
+            return -1;
+        }
+        let lc = post.getElementsByClassName("likesCount")[0];
+        for (let i = 0; i < photoPosts.length; i++) {
+            if (photoPosts[i].id == id) {
+                lc.innerHTML = photoPosts[i].likes.size.toString();
+                break;
+            }
+        }
+    }
+
+    function getTime(createdAt) {
+        let date = new Date(createdAt);
+        let days = date.getDate();
+        if (days < 10) {
+            days = '0' + days;
+        }
+        let months = date.getMonth() + 1;
+        if (months < 10) {
+            months = '0' + months;
+        }
+        let years = date.getFullYear();
+        let hours = date.getHours();
+        if (hours < 10) {
+            hours = '0' + hours;
+        }
+        let minutes = date.getMinutes();
+        if (minutes < 10) {
+            minutes = '0' + minutes;
+        }
+        return days + '.' + months + '.' + years + ' ' + hours + ':' + minutes;
```

Comment:
It could be interesting for you to read about template literals in js
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
Everything corrected. But the updated version lies in the Task6 pull request with Task6 code.